lib/mix/tasks/gen.ex

```diff
@@ -189,7 +189,9 @@ defmodule Mix.Tasks.Gen do
     defp create_output(assigns) do
       case MixTemplates.generate(assigns.template_module, assigns) do
         { :error, reason } ->
-          Mix.shell.info([ :red, "Error: ", :reset, reason ])
+          Mix.raise("""
+          Failed to create output for #{assigns.template_module} because #{reason}
+          """)
```

Comment:
Raising a `Mix.Error` to stop the task overall. 
I looked into returning the error tuple and propagating that through `generate_with_options` and `maybe_invoke_based_on` but decided to start with this less invasive change.